mayastor/src/bdev/nexus/nexus_io.rs

```diff
@@ -348,6 +290,65 @@ impl Bio {
         }
     }
 
+    /// assess the IO if we need to mark it failed or ok.
+    #[inline]
+    pub(crate) fn assess(&mut self, child_io: &mut Bio, success: bool) {
+        self.ctx_as_mut_ref().dec();
+
+        if !success {
+            // currently, only tests send those but invalid op codes should not
+            // result into faulting a child device.
+            if NvmeStatus::from(child_io.clone()).status_code()
+                == GenericStatusCode::InvalidOpcode
+            {
+                self.complete();
+                return;
+            }
+
+            // all other status codes indicate a fatal error
+            Reactors::master().send_future(Self::child_retire(
+                self.nexus_as_ref().name.clone(),
+                child_io.bdev_as_ref(),
+            ));
+        }
+
+        self.complete();
+    }
+
+    async fn child_retire(nexus: String, child: Bdev) {
+        error!("{:#?}", child);
+
+        if let Some(nexus) = nexus_lookup(&nexus) {
+            if let Some(child) = nexus.child_lookup(&child.name()) {
+                let current_state = child.state.compare_and_swap(
+                    ChildState::Open,
+                    ChildState::Faulted(Reason::IoError),
+                );
+
+                if current_state == ChildState::Open {
+                    warn!(
+                        "core {} thread {:?}, faulting child {}",
+                        Cores::current(),
+                        Mthread::current(),
+                        child,
+                    );
+
+                    let uri = child.name.clone();
+                    nexus.pause().await.unwrap();
+                    nexus.reconfigure(DREvent::ChildFault).await;
+                    //nexus.remove_child(&uri).await.unwrap();
+                    bdev_destroy(&uri).await.unwrap();
+                    if nexus.status() != NexusStatus::Faulted {
+                        nexus.resume().await.unwrap();
+                    } else {
+                        error!(":{} has no children left... ", nexus);
```

Comment:
If the nexus is faulted, this means we stay in the "paused" state. Is this ok? I assume so as at this point the nexus is unrecoverable.

Reply:
Yes, for now, we have no children left so we cant turn it back on...